draft-dcook-ppm-dap-interop-test-design.md

```diff
@@ -136,12 +136,12 @@ The test runner will POST an empty object (i.e. `{}`) to this endpoint to check
 
 Request the base URL for DAP-PPM endpoints for a new task. This API will be invoked immediately before `/internal/test/add_task` (see {{aggregator-add-task}}), to determine the endpoint URLs of the aggregators. If the aggregator uses a common set of DAP-PPM endpoints for all tasks, it could always return the same value, such as the relative URL `/`. Alternately, implementations may wish to generate new endpoints for each task, derive the endpoint based on the `TaskId`, etc.
 
-The test runner will provide the hostname and port at which the aggregator is externally reachable. If the aggregator returns a relative URL, the test runner will combine it with the hostname and port into an absolute URL. Otherwise, the aggregator can incorporate the hostname and port into an absolute URL and return that.
+The test runner will provide the hostname at which the aggregator is externally reachable. If the aggregator returns a relative URL, the test runner will combine it with the hostname into an absolute URL, assuming that the port is 8080. Otherwise, the aggregator can incorporate the hostname into an absolute URL and return that.
 
 |Key|Value|
 |`taskId`|A base64url-encoded DAP-PPM `TaskId`|
 |`aggregatorId`|0 if this aggregator is the leader, or 1 if this aggregator is the helper.|
-|`hostnameAndPort`|This aggregator's hostname and port in the interoperation test environment. This may optionally be used in constructing the endpoint URL as an absolute URL.|
+|`hostname`|This aggregator's hostname in the interoperation test environment. This may optionally be used in constructing the endpoint URL as an absolute URL.|
```

Comment:
Note: I dropped the port from this because I suppose that a container can assume it will be reached on port 8080 by default -- even if the port is exposed to the outside world, from the container's POV it will be serving the test API from port 8080.

OTOH, the DAP serving port (if different) isn't necessarily mapped to the outside world at all. That's OK as long as the test runner sets up container networking such that all of the containers can talk to each other (in Docker parlance, they'd all part of the same bridge network, I suppose).

I think the networking isn't called out, but it would make a lot of sense (IMO) to run tests this way -- it allows containers not to have to worry about the Container<->Host port mapping, and improves isolation. WDYT?

Reply:
Sounds good to me. One of the reasons I included the port originally was as a hedge for future support for reverse proxies/network simulators in front of each container. (probably in yet more containers in the same network) The test runner can instead just parse ports out of URLs and update hosts and ports as needed, both for that case, and for forwarding to the test runner itself on the host via Docker's EXPOSE.